dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectFactoryImpl.java

```diff
@@ -57,9 +57,20 @@ public CDIManagedObjectFactoryImpl(Class<T> classToManage, CDIRuntime cdiRuntime
 
     @Override
     public ManagedObject<T> createManagedObject() throws ManagedObjectException {
+        isWeldProxy();
```

Comment:
Why check the class here and not at the point where the MOF is constructed (where the class is passed in).

Reply:
And I am pretty sure this would apply across all flavours of the MOF so the check can be done in the constructor of AbstractManagedObjectFactory